orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/Message.kt

```diff
@@ -24,12 +24,50 @@ import com.netflix.spinnaker.orca.Task
 import com.netflix.spinnaker.orca.pipeline.model.Execution
 import com.netflix.spinnaker.orca.pipeline.model.Stage
 import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
+import java.util.*
 
 /**
  * Messages used internally by the queueing system.
  */
+
+@JsonTypeInfo(use = MINIMAL_CLASS, include = PROPERTY, property = "@class")
+interface Attribute {
+}
+
+data class MaxAttemptsAttribute(val maxAttempts: Int = -1) : Attribute {
+}
+
+data class AttemptsAttribute(var attempts: Int = 0) : Attribute {
+  fun increment() {
+    this.attempts = attempts + 1
+  }
+}
+
 @JsonTypeInfo(use = MINIMAL_CLASS, include = PROPERTY, property = "@class")
-sealed class Message
+sealed class Message {
+  val attributes: MutableList<Attribute> = mutableListOf()
```

Comment:
Why a List? Seems like this is a map-like structure (although see comments elsewhere about how we could just make it completely typed).

I'm concerned this makes the message type mutable rather than using a `copy` operation to increment counts.